actionpack/lib/action_dispatch/system_test_case.rb

```diff
@@ -105,9 +105,12 @@ def self.start_application # :nodoc:
     #
     #   driven_by :selenium, screen_size: [800, 800]
     def self.driven_by(driver, using: :chrome, screen_size: [1400, 1400])
-      SystemTesting::Driver.new(driver).run
+      driver = selenium?(driver) ? SystemTesting::Browser.new(using, screen_size) : SystemTesting::Driver.new(driver)
+
+      setup { driver.use }
+      teardown { driver.reset }
```

Comment:
Which is fine, I don't know of another way right now and we'll have to see if there is a performance impact. I thought that Rspec config.before/config.after ran on the suite and only ran on each test if you wrote `config.before(:each)`/`config.after(:each)` - but I don't use Rspec so I may be wrong on that front. This is fine for now.